examples/glnn/train_student.py

```diff
@@ -0,0 +1,173 @@
+
+    def forward(self, data, teacher_logits):
+        student_logits = self.backbone_network(data['x'])
+        train_y = tlx.gather(data['y'], data['t_idx'])
+        train_teacher_logits = tlx.gather(teacher_logits, data['t_idx'])
+        train_student_logits = tlx.gather(student_logits, data['t_idx'])
+        loss = self._loss_fn(train_y, train_student_logits, train_teacher_logits, args.lamb)
+        return loss
+
+
+def get_training_config(config_path, model_name, dataset):
+    with open(config_path, "r") as conf:
+        full_config = yaml.load(conf, Loader=yaml.FullLoader)
+    dataset_specific_config = full_config["global"]
+    model_specific_config = full_config[dataset][model_name]
+
+    if model_specific_config is not None:
+        specific_config = dict(dataset_specific_config, **model_specific_config)
+    else:
+        specific_config = dataset_specific_config
+
+    specific_config["model_name"] = model_name
+    return specific_config
+
+
+def calculate_acc(logits, y, metrics):
+    metrics.update(logits, y)
+    rst = metrics.result()
+    metrics.reset()
+    return rst
+
+
+def kl_divergence(teacher_logits, student_logits):  
+    # convert logits to probabilities
+    teacher_probs = tlx.softmax(teacher_logits)
+    student_probs = tlx.softmax(student_logits)
+    # compute KL divergence
+    kl_div = tlx.reduce_sum(teacher_probs * (tlx.log(teacher_probs+1e-10) - tlx.log(student_probs+1e-10)), axis=-1)
+    return tlx.reduce_mean(kl_div)
+
+
+def cal_mlp_loss(labels, student_logits, teacher_logits, lamb):
+    loss_l = tlx.losses.softmax_cross_entropy_with_logits(student_logits, labels)
+    loss_t = kl_divergence(teacher_logits, student_logits)
+    return lamb * loss_l + (1 - lamb) * loss_t
+
+
+def train_student(args):
+    # load datasets
+    if str.lower(args.dataset) not in ['cora','pubmed','citeseer','computers','photo']:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    if args.dataset in ['cora', 'pubmed', 'citeseer']:
+        dataset = Planetoid(args.dataset_path, args.dataset)
+    elif args.dataset in ['computers', 'photo']:
+        dataset = Amazon(args.dataset_path, args.dataset)
+    graph = dataset[0]
+
+    # load teacher_logits from .npy file
+    teacher_logits = tlx.files.load_npy_to_any(path = r'./', name = f'{args.dataset}_{args.teacher}_logits.npy')
+    teacher_logits = tlx.ops.convert_to_tensor(teacher_logits)
+
+    # for mindspore, it should be passed into node indices
+    if args.dataset in ['cora', 'pubmed', 'citeseer']:
+        train_idx = mask_to_index(graph.train_mask)
+        test_idx = mask_to_index(graph.test_mask)
+        val_idx = mask_to_index(graph.val_mask)
+        t_idx = tlx.concat([train_idx, test_idx, val_idx], axis=0)
+    elif args.dataset in ['computers', 'photo']:
+        train_mask, val_mask, test_mask = get_train_val_test_split(dataset, train_per_class=20, val_per_class=30)
```

Comment:
Normally, you may get the mask through the `dataset.train_mask`. So, it would be better to modify the `gammagl.dataset.Amazon`.

Reply:
The requested modifications have been made to train_teacher.py, train_student.py, amazon.py, and get_split.py. 
Thank you for the reminder!